### PR TITLE
Numpydoc for backends

### PR DESCRIPTION
**Pull request by QuLogic**
I've had this lying around for a while, but it's incomplete insofar as I didn't have a chance to add types to everything. I don't really want it to go completely to waste though.

Apparently, I had a lot of free time at the beginning of the year and did the same for a _lot_ of other files. The trouble is, that was with the expectation of switching to napoleon, so there are a few corrections that need to be made to work with our current build (which you can see in the second commit here.) Should I post those up somewhere (and maybe tag with Need contributor)?

### CODE REVIEW

**Comment by QuLogic** [comment]
I've added types to all the backend files I've already touched. The descriptions are not perfect, but several of the backends do not appear in the documentation, presumably because they cannot all be loaded at the same time or the dependencies are not available. I tried adding them, but it broke the builds, so that's something for another time.

File: lib/matplotlib/backend_bases.py [hunk 1]
```diff
@@ -1189,19 +1219,20 @@ class TimerBase(object):
...
 
-    Attributes:
+    Attributes
+    ----------
+    interval : scalar
+        The time between timer events in milliseconds. Default is 1000 ms.
 
-        * `interval`: The time between timer events in
-          milliseconds. Default is 1000 ms.
+    single_shot: bool
```
Comment (efiring) [inline lib/matplotlib/backend_bases.py]: Here and below, my understanding from @nellev is that the colon must be surrounded by spaces.

Comment (QuLogic) [inline lib/matplotlib/backend_bases.py]: Grrr, thought I got all of those.

Comment (NelleV) [inline lib/matplotlib/backend_bases.py]: Yep, else it won't render properly. I wouldn't hold this PR for this thought. It's been around way too long.

**Comment by QuLogic** [comment]
Huh, that was way more outdated than I thought, but fortunately no conflicts. Rebased and addressed the last two comments.

**Pull request merged**